### PR TITLE
Remove `mem-ballast-size-mib` from command line options

### PR DESCRIPTION
**Pull request by mxiamxia**
**Description:** <Describe what has changed. 
* Remove `mem-ballast-size-mib` from command line options
* Replace `mem-ballast-size-mib` setting by `ballast extension` in testings

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector/issues/2516

### CODE REVIEW

File: examples/k8s/otel-config.yaml [hunk 1]
```diff
@@ -25,8 +25,6 @@ data:
     processors:
       batch:
       memory_limiter:
-        # Same as --mem-ballast-size-mib CLI argument
-        ballast_size_mib: 165
```
Comment (bogdandrutu) [inline examples/k8s/otel-config.yaml:29]: Changes in this file can be split in an independent PR.

File: testbed/tests/e2e_test.go [hunk 2]
```diff
@@ -18,7 +18,9 @@
 package tests
 
 import (
-	"strconv"
```
Comment (bogdandrutu) [inline testbed/tests/e2e_test.go:21]: Changes in the `testbed` package can be split in an independent PR.

**Review by bogdandrutu — COMMENTED** [review]
Let's try to create more focused and simple PRs :)

**Comment by mxiamxia** [comment]
Close and split this PR to 2 PRs
